rfc/rfc018-openid-credential-issuance.md

```diff
@@ -0,0 +1,140 @@
+# RFC018 Verifiable Credential Issuance using OpenID
+
+|                           |           |
+|:--------------------------|:----------|
+| Nuts foundation           | R.G. Krul |
+| Request for Comments: 018 | Nedap     |
+
+## Verifiable Credential Issuance using OpenID
+
+### Abstract
+
+This RFC describes how to use [OpenID4VCI](https://openid.net/specs/openid-4-verifiable-credential-issuance-1_0.html)
+to issue credentials from a Nuts node, to another Nuts node. It adds security requirements, wallet discovery and
+server-to-server responses to make it applicable to Nuts.
+
+### Status of document
+
+This document describes a Nuts standards protocol.
+
+### Copyright Notice
+
+![](../.gitbook/assets/license.png)
+
+This document is released under the [Attribution-ShareAlike 4.0 International \(CC BY-SA 4.0\) license](https://creativecommons.org/licenses/by-sa/4.0/).
+
+## 1. Introduction
+
+This specification describes how to apply OpenID Connect for Verifiable Credential Issuance (OpenID4VCI) to issue Verifiable Credentials (VCs)
+to resolvable DIDs in server-to-server flows, without user (human) interaction from the holder.
+
+## 2. Terminology
+
+This specification uses the terms "DID", "DID document" and "Service" as defined by the [DID Core Data Model](https://www.w3.org/TR/did-core/),
+"Verifiable Credential", "Credential Subject", "Holder" and "Issuer" as defined by the [Verifiable Credential Data Model](https://www.w3.org/TR/vc-data-model/), and
+"Wallet", "Client Metadata" and "Credential Offer" as defined by [OpenID 4 Verifiable Credential Issuance](https://openid.net/specs/openid-4-verifiable-credential-issuance-1_0.html).
+
+## 3. Overview
+
+While OpenID4VCI mainly describes issuing credentials through a browser or to a mobile wallet app,
+this specification describes how to use OpenID4VCI to issue credentials server-to-server, without user interaction from the holder.
+
+Nodes MUST support both processes of issuing and receiving credentials over OpenID4VCI.
+
+## 3.1 Supported Flows and Features
+
+Conforming implementations MUST support the following features:
+
+- Pre-authorized code flow
+- Credential offer:
+  - with `credential_offer` parameter
+  - with `string` values, referring to a credential's `id` in `credentials_supported`
+- Credentials in JSON-LD format
+- Issuer- and wallet metadata
+
+## 3.2 Credential Offer Response
+
+The OpenID4VCI specification does not specify responses for the credential offer.
+But since this specification describes server-to-server interaction
+a response is useful for the issuer to know whether the credential offer was accepted.
+
+### 3.2.1 Credential Offer Success Response
+
+In case the wallet considers the offer valid, it MUST respond with HTTP status `200 OK`.
+This is regardless of whether the wallet will actually revrieve the wallet.
+
+If the wallet decides to retrieve the credential synchronously, it SHOULD respond with `status` set to `credential_received`.
```

Comment:
OpenID4VCI spec has renamed the **credential request** `synchronous` flow to `immediate issuance` flow. We probably want to use the same terminology? But I do not know how/if this applies to the offer...

Reply:
It's a slightly different thing here, from the wallet's perspective. This is not spec'd. But I renamed it to be more like the spec.